### ISSUE TITLE
DAS and FCA training consult

### ISSUE THREAD

**Issue by mamougo** [winterwindwang/Full-coverage-camouflage-adversarial-attack#27]
Hi!
   I'm very interested in DAS and your work and trying to train the camouflage, but I got some problems with the code. I pulled your neural renderer edition and downloaded the data, but when I trained the DAS camouflage I got NAN output in the mask_renderer forward step. ![QQ截图20221024214214](https://user-images.githubusercontent.com/26672076/197540093-27455f28-862d-48e4-8512-f72fb24d43be.png) and the loss looks like that ![image](https://user-images.githubusercontent.com/26672076/197540480-fa99b54a-dd9b-4e5b-95bd-52b1b735f69b.png).
  I was wondering is that correct?  or I did something wrong? And in the FCA method do I need to generate the training image from ,npz file first and then run the train.py?
  My env is as follows:
GTX 3080ti
cuda 11.6
python 3.8
torch 11.6
   Looking forward to your reply!

**Comment by mamougo** [comment]
Sorry, the mask_renderer.forward step is got imgs_pred all zero.

**Comment by winterwindwang** [comment]
> Sorry, the mask_renderer.forward step is got imgs_pred all zero.

Check the rendered image whether contain content (i.e., car). I guess the rendered image has nothing except pure color background. If you encounter this issues, please contact me futher.

**Comment by mamougo** [comment]
> 

Yes, the test_total.jpg is all black when the loss is NAN and sometimes the test_total.jpg is a car with black texture on the road.
I just use the training data(12500 .npz file) and texture(das/src/texture) provided by the das author and your neural renderer and nmr_test.
